content/docs/rfcs/43/README.md

```diff
@@ -0,0 +1,347 @@
+
+We note that since the `ChaChaPoly` cipher used to encrypt messages supports *additional data*, 
+an encrypted payload can be further authenticated by passing the `messageNametag` as additional data to the encryption/decryption routine. 
+In this way, an attacker would be unable to craft an authenticated Waku message 
+even in case the currently used symmetric encryption key is compromised, 
+unless `ctsInbound`, `ctsOutbound` or the `messageNametag` buffer lists were compromised too.
+
+# Security/Privacy Considerations
+
+### Assumptions
+
+- The attacker is active, i.e. can interact with both devices `A` and `B` by sending messages over `contentTopic`.
+
+- The attacker has access to the QR code, that is knows the ephemeral key `eB`, the commitment `H(sB||r)` and the `contentTopic` exposed by the device `B`.
+
+- Devices `A` and `B` are considered trusted (otherwise the attacker will simply exfiltrate the relevant information from the attacked device).
+
+- As common for Noise, we assume that ephemeral keys cannot be compromised, while static keys might be later compromised. However, we enforce in the pairing phase extra security mechanisms (i.e. use of commitments for static keys) that will prevent some attacks possible when ephemeral keys are weak or get compromised.
+ 
+### Rationale
+
+- The device `B` exposes a commitment to its static key `sB` because:
+    - if the private key of `eB` is weak or gets compromised, an attacker can impersonate `B` by sending in message `c.` to device `A` his own static key and successfully complete the pairing phase. Note that being able to compromise `eB` is not contemplated by our security assumptions.
+    - `B` cannot adaptively choose a static key based on the state of the Noise handshake at the end of message `b.`, i.e. after the authentication code is confirmed. Note that device `B` is trusted in our security assumptions.
+    - Confirming the authentication code after processing message `b.` will ensure that no Man-in-the-Middle (MitM) can send a static key different than `sB`.
+
+- The device `A` sends a commitment to its static key `sA` because:
+    - `A` cannot adaptively choose a static key based on the state of the Noise handshake at the end of message `b.`, i.e. after the authentication code is confirmed. Note that device `A` is trusted in our security assumptions.
+    - Confirming the authentication code after processing message `b.` will ensure that no MitM can send a static key different than `sA`.
+
+- The authorization code is shown and has to be confirmed at the end of message `b.` because:
+    - an attacker that frontruns device `A` by sending faster his own ephemeral key would be detected before  he's able to know device `B` static key `sB`;
+    - it ensures that no MitM attacks will happen during *the whole* pairing handshake, since commitments to the (later exchanged) device static keys will be implicitly acknowledged by the authorization code confirmation;
```

Comment:
I think there should be more to the commitment idea (which is not explained in the text), as for example, sending `sB` and `r` instead of the commitment `H(sB||r)` at step `a.` would also do the job, because `A` can persist `r` and later checks whether the same `r` is being sent at step `c.`.
The same holds for `H(sA||s)`, i.e., `A` can send `sA` and `s` at step `b.` and later resends `s` in step `d.` so that `B` can do consistency check.  Since the message sent at `b.` is already encrypted, it is protected against eavesdroppers. Please let me know if I am missing something in here. 
In general, I like the idea of commitment, but I think the supplied rationale does not cover all the aspects you had in mind. 
One suggestion is that to enumerate the security objectives of the protocol and then relate each design choice to that objective e.g., 1) MiM secure: the protocol is supposed to be secure against MiM, or 2) Anonymous: the protocol is designed to be anonymous such that the static keys of the sender and the receiver remain unknown throughout the protocol, etc.

Reply:
Right, how would you argue about the part that the static key of Bob is exchanged using QR but not via a p2p network? So, anonymity is preserved by design. I mean, the commitment of `sB` does not seem necessary given that the interaction is entirely offline and face-to-face.

Reply:
If there is any type of attack that can happen in the QR code interaction, that is worth highlighting in the text.